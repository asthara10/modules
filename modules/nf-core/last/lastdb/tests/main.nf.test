nextflow_process {

    name "Test Process LAST_LASTDB"
    script "../main.nf"
    config "./nextflow.config"
    process "LAST_LASTDB"

    tag "modules"
    tag "modules_nfcore"
    tag "last"
    tag "last/lastdb"

    test("test_last_lastdb") {


        when {
            process {
                """
                
                input[0] = [ [ id:'test' ], // meta map
              file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
            ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_last_lastdb_gzipped_input") {
        when {
            process {
                """
                        input[0] = [ [ id:'test' ], // meta map
              file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
            ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
