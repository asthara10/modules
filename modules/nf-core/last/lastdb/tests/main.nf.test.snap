{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d7ebd06314ca501f8781d0c729c71516"
            ]
        ],
        "timestamp": "2023-11-24T13:57:25.498069"
    }
}